### PR TITLE
Add testing

### PR DESCRIPTION
**Pull request by DeanoDee**

### CODE REVIEW

File: src/App/index.test.js [hunk 1]
```diff
@@ -1,8 +1,51 @@
...
+		const div = document.createElement('div');
+		ReactDOM.render(<App />, div);
+	});
+	it('changeEndpoint sets the endpoint and loading to true, and clears data and errors', () => {
+		const mockEndpoint = "http://localhost:3002/alternateImages.json";
+		app.changeEndpoint(mockEndpoint);
+		const {endpoint, loading, data, error, subscription} = app.state;
+		expect(endpoint).toBe(mockEndpoint);
+		expect(loading).toBe(true);
+		expect(data).toBe(null);
```
Comment (Mackarous) [inline src/App/index.test.js:19]: why are you expecting `data` to be `null`?

File: src/App/index.test.js [hunk 2]
```diff
@@ -1,8 +1,51 @@
...
+					"is"
+				]
+			}
+		]};
+		app.setData(mockData);
+		const {data, loading} = app.state;
+		expect(data).toBe(mockData.name);
+		expect(loading).toBe(false);
+	});
+	it('setError set and error and loading to true', () => {
```
Comment (Mackarous) [inline src/App/index.test.js]: > "set and error and loading to true"

Not sure what that means

**Review by Mackarous — CHANGES_REQUESTED** [review]
Pretty cool test lib!

File: src/App/index.test.js (hunk 1 above)
Comment (DeanoDee) [reply src/App/index.test.js:19]: When you trigger change in endpoint we clear the data to prepare for the new data

File: src/App/index.test.js (hunk 2 above)
Comment (DeanoDee) [reply src/App/index.test.js]: Yeah that should be clarified

**Review by Mackarous — APPROVED** [review]

**Pull request merged**